Source/AFError.swift

```diff
@@ -189,13 +198,26 @@ public enum AFError: Error {
     case sessionInvalidated(error: Error?)
     /// `URLRequest` failed validation.
     case urlRequestValidationFailed(reason: URLRequestValidationFailureReason)
+    ///  `UploadableConvertible` threw an error in `createUploadable()`
+    case createUploadableFailed(error: Error)
+    ///  `URLRequestConvertible` threw an error in `asURLRequest()`
+    case createURLRequestFailed(error: Error)
+    /// `SessionDelegate` threw an error while attempting to move downloaded file to destination URL
+    case downloadedFileMoveFailed(error: Error, destination: URL)
+    /// `URLSessionTask` completed with error
+    case sessionTaskFailed(error: Error)
 }
 
 extension Error {
     /// Returns the instance cast as an `AFError`.
     public var asAFError: AFError? {
         return self as? AFError
     }
+
+    /// Casts the instance as `AFError` or returns `defaultAFError`
+    func asAFError(default defaultAFError: @autoclosure () -> AFError) -> AFError {
```

Comment:
I liked the previous `or` external label here: `func asAFError(or makeAFError:)`, as it reads better on usage: `error.asAFError(or: AFError.`.

Reply:
Done.